File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {
        long sum = 0; // long으로 변경
        for(int i = 1; i <= count; i++){
            sum += (long)price * i; // 곱셈도 long으로 안전하게
        }
        long a = sum - money; // long으로 선언
        return a > 0 ? a : 0; // 돈이 충분하면 0 반환
    }
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.js
function solution(price, money, count) {
    var answer = 0;
    let sum = 0;
    while(count != 0){
        sum += price * count;
        count--;
    }
    if(sum <= money){
        answer = 0;
    }else{
         answer = sum - money
    }
   

    return answer;
}